pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/prometheus/common/model"
+)
+
+var InsecureRoundTripper http.RoundTripper = &http.Transport{
+	Proxy: http.ProxyFromEnvironment,
+	DialContext: (&net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}).DialContext,
+	TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
+	TLSHandshakeTimeout: 10 * time.Second,
+}
+
+type TimePoint [2]float64
+type TimeSeriesPoints []TimePoint
+
+type TimeSeries struct {
+	Name   string
+	Points TimeSeriesPoints
+	Tags   []rawData
+}
+
+type rawData struct {
+	Name   string
+	Values []float64
+}
+
+type prometheusConfig struct {
+	config api.Config
+	API    v1.API
+	Errors []error
+}
+
+func newPrometheusConfig(url, token string, insecureTLS bool) prometheusConfig {
+	return prometheusConfig{config: api.Config{Address: url, Authorization: token, RoundTripper: InsecureRoundTripper}}
+}
+
+func (c *prometheusConfig) newPrometheusAPI() {
+	client, err := api.NewClient(c.config)
+	if err != nil {
+		fmt.Printf("Error creating Prometheus client: %v\n", err)
+		c.Errors = append(c.Errors, err)
+	} else {
+		c.API = v1.NewAPI(client)
+	}
+}
+
+// DoPrometheusQuery will run queries against Prometheus endpoint
+func DoPrometheusQuery(cfg config.ScrapeConfig) {
+	//cfg.UrlFlag, cfg.TokenFlag, cfg.StepFlag, cfg.DurationFlag, cfg.InsecureTLSFlag
+	// url, token, stepFlag string, durationFlag int, insecureTLS bool
+	config := newPrometheusConfig(cfg.UrlFlag, cfg.TokenFlag, cfg.InsecureTLSFlag)
+	config.newPrometheusAPI()
+	if len(config.Errors) != 0 {
+		fmt.Printf("Client Error %v\n", config.Errors[0])
+	}
+
+	queries := map[string]string{
+		"cpu":    "sum(rate(container_cpu_usage_seconds_total{job=\"kubelet\", image!=\"\", container!=\"POD\"}[5m])) by (namespace)",
+		"memory": "sum(container_memory_usage_bytes{container_name!=\"\"}) by (namespace)",
+	}
+	end := time.Now()
+	start := end.Add(time.Duration(-1*cfg.DurationFlag) * time.Minute)
+	step, err := time.ParseDuration("1m")
+	if err != nil {
+		fmt.Printf("Error parsing step duration %s\n", cfg.StepFlag)
+	}
+	r := v1.Range{Start: start, End: end, Step: step}
+
+	var results []result.ResultType
+	for resource, query := range queries {
+		queryResult, _, err := config.API.QueryRange(context.Background(), query, r)
+		if err != nil {
+			fmt.Printf("Query Error: %v\n", err)
+			return
+		}
+		fmt.Printf("Query returned: %+v\n", queryResult)
+
+		data, ok := queryResult.(model.Matrix)
+		if !ok {
+			fmt.Printf("Unsupported result format: %s\n", queryResult.Type().String())
+			return
+		}
+
+		series := TimeSeries{
+			Name: query,
+			Tags: []rawData{},
+		}
+
+		for _, j := range data {
+			var newData rawData
+			for _, v := range j.Metric {
+				newData.Name = string(v)
+			}
+			for _, v := range j.Values {
+				newData.Values = append(newData.Values, float64(v.Value))
+			}
+			series.Tags = append(series.Tags, newData)
+		}
+		fmt.Printf("Series: %+v\n", series)
```

Comment:
Is this for debugging or part of output?

Reply:
It's for debugging

Reply:
If it is for debugging, shouldn't debugging info go to stderr?

Reply:
Fixed.